Data/Primitive/SmallArray.hs

```diff
@@ -436,6 +437,20 @@ unsafeTraverseSmallArray f (SmallArray ar) = SmallArray `liftM` unsafeTraverseAr
 #endif
 {-# INLINE unsafeTraverseSmallArray #-}
 
+-- | Strict map over the elements of the array.
+mapSmallArray' :: (a -> b) -> SmallArray a -> SmallArray b
+#if HAVE_SMALL_ARRAY
+mapSmallArray' f sa = createSmallArray (length sa) (die "mapSmallArray'" "impossible") $ \smb ->
+  fix ? 0 $ \go i ->
+    when (i < length sa) $ do
```

Comment:
History, I imagine. The `fix` seems to avoid the need for a local type signature in the `go` function. I have no idea how the generated code compares, but that would be worth checking.

Reply:
I'm inclined to leave this as is for now. I'd like to open a separate PR adding benchmarks that check the difference between these two styles.